osu.Game/Screens/Play/HitErrorDisplay/DefaultHitErrorDisplay.cs

```diff
@@ -0,0 +1,178 @@
+            : base(overallDifficulty, hitWindows)
+        {
+            maxHitWindows = HitWindows.Meh == 0 ? HitWindows.Good : HitWindows.Meh;
+
+            AutoSizeAxes = Axes.Both;
+            AddInternal(new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.X,
+                Height = bar_height,
+                Direction = FillDirection.Horizontal,
+                Spacing = new Vector2(spacing, 0),
+                Children = new Drawable[]
+                {
+                    judgementsContainer = new Container
+                    {
+                        Anchor = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Origin = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Width = judgement_line_width,
+                        RelativeSizeAxes = Axes.Y,
+                    },
+                    bar = new FillFlowContainer
+                    {
+                        Anchor = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Origin = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Width = bar_width,
+                        RelativeSizeAxes = Axes.Y,
+                        Direction = FillDirection.Vertical,
+                    },
+                    new Container
+                    {
+                        Anchor = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        Origin = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
+                        AutoSizeAxes = Axes.X,
+                        RelativeSizeAxes = Axes.Y,
+                        Child = arrow = new SpriteIcon
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativePositionAxes = Axes.Y,
+                            Icon = reversed ? FontAwesome.Solid.ChevronRight : FontAwesome.Solid.ChevronLeft,
+                            Size = new Vector2(8),
+                        }
+                    },
+                }
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            Box topGreenBox;
+            Box bottomGreenBox;
+
+            if (HitWindows.Meh != 0)
+                bar.Add(new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = ColourInfo.GradientVertical(colours.Yellow.Opacity(0), colours.Yellow),
+                    Height = (float)((maxHitWindows - HitWindows.Good) / (maxHitWindows * 2))
+                });
+
+            bar.AddRange(new Drawable[]
+            {
+                topGreenBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colours.Green,
+                    Height = (float)((HitWindows.Good - HitWindows.Great) / (maxHitWindows * 2))
```

Comment:
what is all this division logic? why `maxHitWindows * 2`?

Reply:
Can't really tell much. I was playing with values and this solution is best so far (I mean accurate). May be there's more efficient way to do that